packages/teams-js/test/public/authentication.spec.ts

```diff
@@ -90,647 +412,480 @@ describe('authentication', () => {
+    it('authentication.getUser should not allow calls before initialization', () => {
+      expect(() => authentication.getUser()).toThrowError('The library has not yet been initialized');
     });
 
-    await utils.initializeWithContext('authentication');
+    Object.values(FrameContexts).forEach(context => {
+      it(`authentication.getUser should successfully get user profile in legacy flow with ${context} context`, done => {
+        utils.initializeWithContext(context).then(() => {
+          const successCallback = (user: authentication.UserProfile): void => {
+            expect(user).toEqual(mockResult);
+            done();
+          };
+          const failureCallback = (reason: string): void => {
+            done();
+          };
+          const userRequest: authentication.UserRequest = {
+            successCallback: successCallback,
+            failureCallback: failureCallback,
+          };
+          authentication.getUser(userRequest);
+          const message = utils.findMessageByFunc('authentication.getUser');
+          expect(message).not.toBeNull();
+          expect(message.id).toBe(1);
+          expect(message.args[0]).toBe(undefined);
+          utils.respondToMessage(message, true, mockResult);
+        });
+      });
+
+      it(`authentication.getUser should throw error in getting user profile in legacy flow with ${context} context`, done => {
+        utils.initializeWithContext(context).then(() => {
+          const successCallback = (user: authentication.UserProfile): void => {
+            done();
+          };
+          const failureCallback = (reason: string): void => {
+            expect(reason).toBe(mockResult);
+            done();
+          };
+          const userRequest: authentication.UserRequest = {
+            successCallback: successCallback,
+            failureCallback: failureCallback,
+          };
+          authentication.getUser(userRequest);
+          const message = utils.findMessageByFunc('authentication.getUser');
+          expect(message).not.toBeNull();
+          expect(message.id).toBe(1);
+          expect(message.args[0]).toBe(undefined);
+          utils.respondToMessage(message, false, mockResult);
+        });
+      });
 
-    authentication.notifySuccess(
-      null,
-      'https%3A%2F%2Foutlook.office.com%2Fconnectors%3Fclient_type%3DWin32_Outlook%23%2Fconfigurations',
-    );
-    expect(windowAssignSpyCalled).toBe(true);
-  });
+      it(`authentication.getUser should throw error in getting user profile with ${context} context`, async () => {
+        await utils.initializeWithContext(context);
+        const promise = authentication.getUser();
+        const message = utils.findMessageByFunc('authentication.getUser');
+        expect(message).not.toBeNull();
+        expect(message.id).toBe(1);
+        expect(message.args[0]).toBe(undefined);
+        utils.respondToMessage(message, false, mockResult);
```

Comment:
I'm a bit confused by these couples tests. The test name in 620 and 631 says that getUser should throw error and also successfully execute in the same context. The only difference I see is the difference in respondToMessage. Should these tests be named something like "getUser should ____ with ____ context when _____"? (Although I'm still not sure I understand what the when _____ is in either of these)

Reply:
In respondToMessage we set it to false and promise throws the error. Where as in 631 test cases it resolves the error.